examples/mamba/convert_hf_to_nanotron.py

```diff
@@ -0,0 +1,342 @@
+        def _get_tensor(path: str):
+            param = ref_module.get_parameter(path)
+            return param.grad
+    
+    param = _get_tensor(hf_name)
+    
+    if "in_proj" in hf_name:
+        # In Nanotron, we do tensor parallel column so weight need to be split in the column dimension (i.e: xz.view(...))
+        # However, the HF weights was trained such that it expected xz.chunk(...) to split the tensor in the row dimension
+        # Thus, we need to interleaved the HF weights to make it compatible with Nanotron
+        log_rank(f"Interleaving {hf_name} to make it compatible with Nanotron", logger=logger, level=logging.INFO, rank=0)
+        param = param[_interleave_pattern(param.shape[0]), :]
+        
+    return param
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Convert HF weights from states-space repo to brrr weights")
+    parser.add_argument("--model", type=str, default="130M", help="130M | 370M | 790M | 1.4B | 2.8B")
+    parser.add_argument("--save_path", type=str, default="mamba-nanotron")
+    parser.add_argument("--dp", type=int, default=1)
+    parser.add_argument("--pp", type=int, default=1)
+    parser.add_argument("--tp", type=int, default=1)
+    args = parser.parse_args()
+
+    if args.model not in ["130M", "370M", "790M", "1.4B", "2.8B"]:
+        raise ValueError("Model should be one of 130M, 370M, 790M, 1.4B, 2.8B")
+
+    save_path = Path(args.save_path)
+
+    parallel_config = ParallelismArgs(
+        dp=args.dp,
+        pp=args.pp,
+        tp=args.tp,
+        pp_engine=AllForwardAllBackwardPipelineEngine(),
+        tp_mode=TensorParallelLinearMode.ALL_REDUCE,
+        tp_linear_async_communication=False,
+    )
+
+    parallel_context = ParallelContext(
+        data_parallel_size=parallel_config.dp,
+        pipeline_parallel_size=parallel_config.pp,
+        tensor_parallel_size=parallel_config.tp,
+    )
+
+    # Set log log levels
+    logging_config = LoggingArgs(
+        log_level="info",
+        log_level_replica="info",
+    )
+
+    # Set log levels
+    set_ranks_logging_level(parallel_context=parallel_context, logging_config=logging_config)
+
+    d_model = None
+    num_hidden_layers = None
+    pretrained_model_name = None
+
+    if args.model == "130M":
+        d_model = 768
+        num_hidden_layers = 24
+        pretrained_model_name = "state-spaces/mamba-130m"
+    elif args.model == "370M":
+        d_model = 1024
+        num_hidden_layers = 48
+        pretrained_model_name = "state-spaces/mamba-370m"
+    elif args.model == "790M":
+        d_model = 1536
+        num_hidden_layers = 24
+        pretrained_model_name = "state-spaces/mamba-790m"
+    elif args.model == "1.4B":
+        d_model = 2048
+        num_hidden_layers = 48
+        pretrained_model_name = "state-spaces/mamba-1.4b"
+    elif args.model == "2.8B":
+        d_model = 2560
+        num_hidden_layers = 64
+        pretrained_model_name = "state-spaces/mamba-2.8b"
```

Comment:
here as well. I think it's better to just specify the model_name. And then load its config from transformers, then load the d_model and num_hidden_layers..